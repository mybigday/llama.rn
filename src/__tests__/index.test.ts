import { initLlama, releaseAllLlama } from '..'
import type { TokenData } from '..'

jest.mock('..', () => require('../../jest/mock'))

Math.random = () => 0.5

test('Mock', async () => {
  const context = await initLlama({
    model: 'test.gguf',
  })
  expect(context.id).toBe(0)
  const events: TokenData[] = []
  const completionResult = await context.completion({
    prompt: 'Test',
  }, data => {
    events.push(data)
  })
  expect(events).toMatchSnapshot()
  expect(completionResult).toMatchSnapshot('completion result')

  expect(await context.bench(512, 128, 1, 3)).toMatchSnapshot('bench')

  await context.initMultimodal({
    path: 'mmproj-test.gguf',
  })
  expect(await context.isMultimodalEnabled()).toBe(true)

  await context.releaseMultimodal()
  expect(await context.isMultimodalEnabled()).toBe(false)

  await context.initVocoder({
    path: 'vocoder-test.gguf',
  })
  expect(await context.isVocoderEnabled()).toBe(true)

  await context.releaseVocoder()
  expect(await context.isVocoderEnabled()).toBe(false)

  await context.release()
  await releaseAllLlama()
})
