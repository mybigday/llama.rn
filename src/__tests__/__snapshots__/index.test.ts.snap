// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mock 1`] = `
Array [
  Object {
    "completion_probabilities": Array [
      Object {
        "prob": 0.9658700227737427,
        "tok_str": " *",
      },
      Object {
        "prob": 0.021654844284057617,
        "tok_str": " Hi",
      },
      Object {
        "prob": 0.012475099414587021,
        "tok_str": " Hello",
      },
    ],
    "token": " *",
  },
  Object {
    "completion_probabilities": Array [
      Object {
        "prob": 0.5133139491081238,
        "tok_str": "g",
      },
      Object {
        "prob": 0.3046242296695709,
        "tok_str": "ch",
      },
      Object {
        "prob": 0.18206188082695007,
        "tok_str": "bl",
      },
    ],
    "token": "g",
  },
  Object {
    "completion_probabilities": Array [
      Object {
        "prob": 0.9886618852615356,
        "tok_str": "igg",
      },
      Object {
        "prob": 0.008458126336336136,
        "tok_str": "ig",
      },
      Object {
        "prob": 0.002879939740523696,
        "tok_str": "reet",
      },
    ],
    "token": "igg",
  },
  Object {
    "completion_probabilities": Array [
      Object {
        "prob": 1,
        "tok_str": "les",
      },
      Object {
        "prob": 1.8753286923356427e-8,
        "tok_str": "ling",
      },
      Object {
        "prob": 3.312444318837038e-9,
        "tok_str": "LES",
      },
    ],
    "token": "les",
  },
  Object {
    "completion_probabilities": Array [
      Object {
        "prob": 1,
        "tok_str": "*",
      },
      Object {
        "prob": 4.459857905203535e-8,
        "tok_str": "*.",
      },
      Object {
        "prob": 3.274198334679568e-8,
        "tok_str": "**",
      },
    ],
    "token": "*",
  },
]
`;

exports[`Mock: bench 1`] = `
Object {
  "modelDesc": "test 3B Q4_0",
  "modelNParams": 2779683840,
  "modelSize": 1600655360,
  "ppAvg": 16.211304,
  "ppStd": 0.021748,
  "tgAvg": 38.570646,
  "tgStd": 1.1958,
}
`;

exports[`Mock: completion result 1`] = `
Object {
  "audio_tokens": Array [
    1000,
    1001,
    1002,
    1003,
    1004,
    1005,
    1006,
    1007,
    1008,
    1009,
    1010,
  ],
  "completion_probabilities": Array [
    Object {
      "content": " *",
      "probs": Array [
        Object {
          "prob": 0.9658700227737427,
          "tok_str": " *",
        },
        Object {
          "prob": 0.021654844284057617,
          "tok_str": " Hi",
        },
        Object {
          "prob": 0.012475099414587021,
          "tok_str": " Hello",
        },
      ],
    },
    Object {
      "content": "g",
      "probs": Array [
        Object {
          "prob": 0.5133139491081238,
          "tok_str": "g",
        },
        Object {
          "prob": 0.3046242296695709,
          "tok_str": "ch",
        },
        Object {
          "prob": 0.18206188082695007,
          "tok_str": "bl",
        },
      ],
    },
    Object {
      "content": "igg",
      "probs": Array [
        Object {
          "prob": 0.9886618852615356,
          "tok_str": "igg",
        },
        Object {
          "prob": 0.008458126336336136,
          "tok_str": "ig",
        },
        Object {
          "prob": 0.002879939740523696,
          "tok_str": "reet",
        },
      ],
    },
    Object {
      "content": "les",
      "probs": Array [
        Object {
          "prob": 1,
          "tok_str": "les",
        },
        Object {
          "prob": 1.8753286923356427e-8,
          "tok_str": "ling",
        },
        Object {
          "prob": 3.312444318837038e-9,
          "tok_str": "LES",
        },
      ],
    },
    Object {
      "content": "*",
      "probs": Array [
        Object {
          "prob": 1,
          "tok_str": "*",
        },
        Object {
          "prob": 4.459857905203535e-8,
          "tok_str": "*.",
        },
        Object {
          "prob": 3.274198334679568e-8,
          "tok_str": "**",
        },
      ],
    },
  ],
  "stopped_eos": true,
  "stopped_limit": false,
  "stopped_word": false,
  "stopping_word": "",
  "text": "*giggles*",
  "timings": Object {
    "predicted_ms": 1330.6290000000001,
    "predicted_n": 5,
    "predicted_per_second": 16.533534140620713,
    "predicted_per_token_ms": 60.48313636363637,
    "prompt_ms": 3805.6730000000002,
    "prompt_n": 5,
    "prompt_per_second": 8.408499626741445,
    "prompt_per_token_ms": 118.92728125000001,
  },
  "tokens_cached": 54,
  "tokens_evaluated": 15,
  "tokens_predicted": 6,
  "truncated": false,
}
`;
